File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        # start with the empty set
        arr = [[]]

        # loop over all of the numbers in the list of numbers 
        for i in range(len(nums)):
            # look at all of the items currently in the list, and add the new number onto each of them 
            for j in range(len(arr)):
                if(arr[j] != None and nums[i] != None):
                    arr.append(arr[j] + [nums[i]])

        return arr

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.rs
impl Solution {
    pub fn find_min(nums: Vec<i32>) -> i32 {
        let mut start = 0;
        let mut end = nums.len()-1; 

        while start < end {
            println!("start: {}, end: {}", start, end);
            let mut mid = start + (end - start) / 2;
            if nums[mid] > nums[end] {
                start = mid + 1;
            } else {
                end = mid;
            }
        }

        nums[start] as i32
    }
}